lib/tire/model/indexing.rb

```
@@ -86,6 +89,15 @@ def mapping(*args)
         def indexes(name, options = {}, &block)
           mapping[name] = options
 
+          unless options[:as].nil?
+            case options[:as]
+              when String, Symbol, Proc
+                true # ok, don't do anything
+              else
+                raise Tire::InvalidAsOptionException.new( options[:as] )
```

To avoid allowing you to define a config that's not supported, just failing early. But I can move this to the to_indexed_hash method too.